gnome-initial-setup/pages/display/gis-display-page.c

```diff
@@ -116,42 +130,22 @@ toggle_overscan (GisDisplayPage *page,
                  gboolean value)
 {
   GisDisplayPagePrivate *priv = gis_display_page_get_instance_private (page);
-  GError *error;
+  gboolean current_value = FALSE;
+  GError *error = NULL;
 
-  if (value == gnome_rr_output_info_get_underscanning  (priv->current_output))
+  if (value == cc_display_monitor_get_underscanning (priv->current_output))
     return;
 
-  gnome_rr_output_info_set_underscanning (priv->current_output, value);
-
-  gnome_rr_config_sanitize (priv->current_config);
-  gnome_rr_config_ensure_primary (priv->current_config);
-
-  error = NULL;
-  gnome_rr_config_apply_persistent (priv->current_config, priv->screen, &error);
+  cc_display_monitor_set_underscanning (priv->current_output, value);
 
-  read_screen_config (page);
-
-  if (error != NULL)
+  if (!cc_display_config_apply (priv->current_config, &error))
     {
       g_warning ("Error applying configuration: %s", error->message);
-      g_error_free (error);
+      g_clear_error (&error);
+      return;
     }
-}
-
-static gboolean
-should_display_overscan (GisDisplayPage *page)
-{
-  GisDisplayPagePrivate *priv = gis_display_page_get_instance_private (page);
-  GnomeRROutput *output;
-  char *output_name;
 
-  output_name = gnome_rr_output_info_get_name (priv->current_output);
-  output = gnome_rr_screen_get_output_by_name (priv->screen, output_name);
-
-  if (!output)
-    return FALSE;
-
-  return gnome_rr_output_supports_underscanning (output);
+  read_screen_config (page);
```

Comment:
Why do you call this explicitly? I would expect we get a changed signal after we set the new configuration, and we'd update it that way.

Reply:
I'm simply preserving the behaviour from the previous implementation, see https://github.com/endlessm/gnome-initial-setup/pull/230/commits/dad0e210cb7ebc4bda80e3586eac19460041f26b#diff-1591211bb91441665f14a68847d53cf9L95

But I agree it might not be needed, I'll try removing it and see how it works

Reply:
It was not needed indeed, removed